### DISCUSSION TITLE
_app.tsx + _document.tsx and fast refresh

### DISCUSSION THREAD

**Discussion by Suniron** [vercel/next.js#17398] [Help] [answered]
Here there,

Does class component break fast refresh? 

In fact, class component are exported by default in _app.tsx and _document.tsx and I have sometimes this error in browser console, after somes changes (not always, but after few minutes of coding): 
```
[Fast Refresh] performing full reload

Fast Refresh will perform a full reload when you edit a file that's imported by modules outside of the React rendering tree.
You might have a file which exports a React component but also exports a value that is imported by a non-React component file.
Consider migrating the non-React component export to a separate file and importing it into both files.

It is also possible the parent component of the component you edited is a class component, which disables Fast Refresh.
Fast Refresh requires at least one parent function component in your React tree.
```

I use stitches in my projet (https://www.npmjs.com/package/@stitches/react) which implant styled components. 

Who can help me to newer show this error message 🙄

Thanks 🙏

**Comment by jamesmosier** [comment, marked as answer]
Hi @Suniron. Fast refresh does not work with class components. Check out the list of limitations for some other caveats: https://nextjs.org/docs/basic-features/fast-refresh#limitations

**Reply by Suniron** [reply]
So, **Fast Refresh** is not compatible with using class components **only** in `_app.tsx` and `_document.tsx`?

How can i use function components in theese files ? I think it's not explained in documentation

**Reply by jamesmosier** [reply]
No, fast refresh does not work in **any** class component.

You cannot use a class component for `_document` (that’s only rendered on the server anyway).

The docs do explain how to use one for `_app` though: https://nextjs.org/docs/advanced-features/custom-app

```js
import App from 'next/app'

function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}

MyApp.getInitialProps = async (appContext) => {
  //calls page's `getInitialProps` and fills `appProps.pageProps`
  const appProps = await App.getInitialProps(appContext);
  return { ...appProps }
}
```

**Reply by Suniron** [reply]
I use also **OvermindJS** in my project but the **with-overmind** example show how to use this with class component (https://github.com/vercel/next.js/tree/canary/examples/with-overmind). 

How can i use function component instead of ?
